style_guides/dart.md

```diff
@@ -0,0 +1,58 @@
+# Shorebird Dart Style Guide
+
+This document outlines the style guide for all Shorebird Dart code.
+
+This guide is a supplement to
+[Effective Dart](https://dart.dev/guides/language/effective-dart) and should be
+read in the same way. If there is a conflict between this guide and Effective
+Dart, this guide takes precedence.
```

Comment:
```suggestion
Dart, this guide takes precedence.

*Note: This style guide only defines rules which are enforced via lint rules.*
```

Reply:
Updated to say:

> This style guide only defines rules which are not enforced via lint rules. If possible, prefer using lint rules to enforce style.

Reply:
Yup sorry I was clearly tired when I reviewed 😴